### PR TITLE
Add Tests for React Post Comment, Post Reply

### PR DESCRIPTION
**Pull request by noi5e**
See #9365 

System tests for:
1. Posting a comment in React
2. Posting a reply to another comment in React

I made a new block in `comment_test.rb` to test run tests for basic comment CRUD functionality against both React notes, and then Rails notes. I reorganized the comment system tests a bit, and left some notes to explain the flow.

---
(This issue is part of the larger Comment Editor Overhaul Project with Outreachy. Refer to Planning Issue #9069 for more context)

### CODE REVIEW

File: test/system/comment_test.rb [hunk 1]
```diff
@@ -60,26 +112,6 @@ def get_path(page_type, path)
       assert_selector("#{'#c' + parent_id_num + 'show'} div div div p", text: comment_response_text)
     end
 
-    test "#{page_type_string}: manual comment and reply to comment" do
-      visit get_path(page_type, nodes(node_name).path)
```
Comment (noi5e) [inline test/system/comment_test.rb:64]: Deleted this test.

**Comment by jywarren** [comment]
Looks like a unit test at `Progress: |======================================================== FAIL NodeTest#test_should_find_all_questions (19.97s)`

https://github.com/publiclab/plots2/pull/9665/checks?check_run_id=2659898221#step:5:36

This seems unrelated. It could be an ordering issue. I'll try restarting it!

**Comment by jywarren** [comment]
Yes, nid 8 was still there but in a different order in the returned array. This may mean we need to add a sort to that test either in the main codebase or in the tests. We could try to resolve this here or in a distinct PR!

File: test/system/comment_test.rb [hunk 2]
```diff
@@ -20,16 +20,68 @@ def get_path(page_type, path)
...
+  #   1. basic CRUD in both React and Rails research notes
+  #   2. tests for research notes
+  #   3. tests for research notes, wikis, and questions
+
+  # PART 1: TESTS FOR BASIC CRUD (REACT & RAILS NOTES)
+  # system tests for BASIC commenting CRUD functionality:
+  #   create (posting comments & replies)
+  #   update (editing comments)
+  #   delete
+  [true, false].each do |is_testing_react|
```
Comment (jywarren) [inline test/system/comment_test.rb:36]: nice code syntax!!

**Review by jywarren — APPROVED** [review]

**Comment by noi5e** [comment]
> Yes, nid 8 was still there but in a different order in the returned array. This may mean we need to add a sort to that test either in the main codebase or in the tests. We could try to resolve this here or in a distinct PR!

@jywarren, There's an issue open for the failed test at #9662. I looked into it further, I think it has something to do with array sorting. Check out [this comment here](https://github.com/publiclab/plots2/issues/9662#issuecomment-847387655).

**Comment by jywarren** [comment]
This should work now that @cesswairimu merged a fix for the upstream issue! Thanks!!!

**Comment by jywarren** [comment]
Rebased!!!

**Comment by jywarren** [comment]
Awesome! Thanks @noi5e !!!

**Comment by noi5e** [comment]
@jywarren Thanks for the rebase!

**Pull request merged**